_extras/guide.md

```diff
@@ -52,7 +53,8 @@ Explain that we use Python because:
 *   It's free.
 *   It has a lot of scientific libraries, and more are constantly being added.
 *   It has a large scientific user community.
-*   It's easier for novices to learn than most of the mature alternatives.
+*   It's easier for novices to learn than most of the mature alternatives, 
+especially due to its syntax that reads like a human language.
```

Comment:
```suggestion
partly due to its syntax that reads more like a human language.
```

Reply:
Hi Naci, that's great you got the changes committed to the branch for the pull request! Regarding this question:

> So I guess whenever a conversation is resolved with some suggestions, that in itself becomes a commit, is that correct?

Clicking the `Resolve conversation` button on a discussion point does not commit anything to your files; it is a mechanism of the GitHub interface to say, "ok, we've made a decision on this and no longer need to discuss it further."

When the maintainers do a review of your pull request, we tend to make comments that suggest ideas for editing your submission, and GitHub allows us to put the comment in-line with the code, so we can all see what we are referencing. Generally this does not automatically update the code, it is giving you our opinion about the code change, and we want you to think about if you want to do this then manually change the file either in the GitHub interface (as you did), or at the command-line using `git`.

It is also possible to make a "suggestion" when doing review of the pull request, where we are actually asking if you want to commit the exact change we are proposing. That is what I have done above with the "partly due to its syntax..." comment, and you should be able to click on the `Commit suggestion` dropdown to commit the changes if you agree with them. In this case of using the `Commit suggestion` button, you do not need to manually edit a file, GitHub will automatically make the commit for you. I believe you used this feature earlier on this pull request. If we did not click on `Commit suggestion` the change would not get automatically committed. If you would like to included the "Suggested change" I have added above this, please try committing the suggestion with the interface above to see how that works. :)

Does that clarify these different aspects of the pull request process?

Reply:
Lauren, thank you very much, I think I have a much better understanding of the process now. I think I have a relatively steep learning curve since I never worked on SW development in teams, but I can say it sure is a very powerful process based on all these elaborate schemes to put all those inputs and interactions into the development.